.github/workflows/build_package.yml

```diff
@@ -150,62 +104,79 @@ jobs:
+
+      - name: Build runtime wheels (Linux)
+        if: "matrix.build_package == 'py-runtime-pkg' && matrix.build_family == 'linux'"
         shell: bash
         run: |
-          export CIBW_BEFORE_BUILD="python -m pip install --upgrade pip>=21.3"
-          python -m cibuildwheel --output-dir bindist ./runtime/
+          packages="iree-runtime iree-runtime-instrumented" \
+          output_dir="./bindist" \
+          ./main_checkout/build_tools/python_deploy/build_linux_packages.sh
 
-      - name: Build runtime wheels (instrumented)
-        if: "matrix.build_package == 'instrumented-py-runtime-pkg'"
+      - name: Build runtime wheels (MacOS)
+        if: "matrix.build_package == 'py-runtime-pkg' && matrix.build_family == 'macos'"
         shell: bash
         run: |
-          export CIBW_BEFORE_ALL_LINUX="./main_checkout/build_tools/github_actions/install_tracy_cli_deps_manylinux2014.sh"
-          export CIBW_BEFORE_BUILD="python -m pip install --upgrade pip>=21.3"
-          # I really dislike how cibuildwheel and GitHub actions make tunneling
-          # environment variables so obtuse. Burn this all with fire.
-          export CIBW_ENVIRONMENT_LINUX="$CIBW_ENVIRONMENT_LINUX IREE_BUILD_TRACY=ON IREE_ENABLE_RUNTIME_TRACING=ON"
-          export CIBW_ENVIRONMENT_MACOS="$CIBW_ENVIRONMENT_LINUX IREE_BUILD_TRACY=ON IREE_ENABLE_RUNTIME_TRACING=ON"
-          export CIBW_ENVIRONMENT_WINDOWS="$CIBW_ENVIRONMENT_LINUX IREE_BUILD_TRACY=ON IREE_ENABLE_RUNTIME_TRACING=ON"
-          python -m cibuildwheel --output-dir bindist ./runtime/
-
-      # Experimental iree.compiler package.
-      - name: Build compiler wheels
-        if: "matrix.build_package == 'py-compiler-pkg'"
+          packages="iree-runtime iree-runtime-instrumented" \
+          output_dir="./bindist" \
+          ./main_checkout/build_tools/python_deploy/build_macos_packages.sh
+
+      ##########################################################################
+      # py-compiler-pkg
+      # Builds the iree-compielr wheel.
+      # One step per OS.
+      ##########################################################################
+      - name: Build compiler wheels (Linux)
+        if: "matrix.build_package == 'py-compiler-pkg' && matrix.build_family == 'linux'"
         shell: bash
         run: |
-          # In pip 21.3, in-tree builds became the default and only way to
-          # build. We require that and make sure to be past that threshold.
-          export CIBW_BEFORE_BUILD="python -m pip install --upgrade pip>=21.3"
-          python -m cibuildwheel --output-dir bindist ./main_checkout/iree/compiler
+          packages="iree-compiler" \
+          output_dir="./bindist" \
+          ./main_checkout/build_tools/python_deploy/build_linux_packages.sh
 
+      - name: Build compiler wheels (MacOS)
+        if: "matrix.build_package == 'py-compiler-pkg' && matrix.build_family == 'macos'"
+        shell: bash
+        run: |
+          packages="iree-compiler" \
```

Comment:
I should probably have flagged this, PACKAGES would be more consistent for env var names. Not blocking.

Reply:
Ack